### ISSUE TITLE
Asynchronous module definition (AMD)

### ISSUE THREAD

**Issue by evanp** [pump-io/pump.io#394]
Client-side modules are enormous and must be debigulated.

**Comment by strugee** [comment]
I don't think we need to do this. #301 should be good enough. Leaving open in case anyone disagrees though.
